new-pipeline/src/dashboards.js

```diff
@@ -433,7 +433,7 @@ function getFilterSetsMapping(filters, comparisonName) {
 function getHumanReadableOptions(filterName, options) {
   var channelVersionOrder = {
     "nightly": 0,
-    "aurora": 1,
+    "dev edition": 1,
```

Comment:
I see, now i understand the problem. This doesn't actually change the way this is displayed, we don't need to change this i think.

The interesting part is near the end of this function, [around here](https://github.com/mozilla/telemetry-dashboard/blob/ead6c6705de571c45fb3be27f16f0a2b29d8f164/new-pipeline/src/dashboards.js#L718).
You could change this to `console.log()` the value of what's returned (or inspect it in the debugger?), then you'll see how this looks and can change it's values.

Reply:
@georgf , I have looked into the code, I have figured out from where the names can be replaced. 
But I am a little confused with the implementation.

The `map` function returns "arrays of length 2 or boolean" in the one-liner function on line 

[https://github.com/mozilla/telemetry-dashboard/blob/ead6c6705de571c45fb3be27f16f0a2b29d8f164/new-pipeline/src/dashboards.js#L718](url)

I was trying to expand the one-liner syntax as : 

`return options.map(function (option) {
        if (option) {
            option == option.replace("/", " ");
            return option;
        }
        else {
            return option == null;
        }
    });`

But it is throwing an error in the end. The error says : 

`uncaught exception: Bad options value: must be an array of arrays, either each with two strings or each with three strings`

![screenshot from 2018-03-05 04-41-49](https://user-images.githubusercontent.com/23060217/36951930-4a2aa566-202f-11e8-9ecb-fe5cd2500962.png)

Hence, I am wondering if both the functions (one-liner and the expanded) are the same.

Can you please help me with this ? Thanks!

Reply:
I'm not @georgf, but I feel like I might be able to help anyway : )

Your expanded syntax (with some whitespace for clarity):
```js
return options.map(function (option) {
  if (option) {
    option == option.replace("/", " ");
    return option;
  } else {
    return option == null;
  } });
```

That first line within the `if (option)` block has two equal signs. It will evaluate to a boolean, and then be discarded. I think you only want a single equal sign.

You could leave this line alone and take the functional programming route and perform a map before this one. Something like

```js
options = options.map(option => /* something about replacing 'aurora' with 'dev edition' */)
return options.map(...this code that already exists);
```

Is this helpful?

Reply:
@chutten , Thank you so much for the help. Actually, first of all I was doing the same thing but caught up in some error. Now, I have fixed it.